oneargopy/Argo.py

```diff
@@ -0,0 +1,279 @@
+# -*- coding: utf-8 -*-
+# Argo.py
+#------------------------------------------------------------------------------
+# Created By: Savannah Stephenson
+# Creation Date: 05/30/2024
+# Version: 1.0
+#------------------------------------------------------------------------------
+""" The Argo class contains the primary functions for downloading and handling
+    data gathered from GDAC.
+"""
+#------------------------------------------------------------------------------
+# 
+#
+# Imports
+
+# Local
+from Settings import DownloadSettings, SourceSettings
+
+# System
+from pathlib import Path
+import pandas as pd
+import time
+import requests
+from datetime import datetime
+import shutil
+import gzip
+import numpy as np
+import logging
+
+class Argo:
+    
+    def __init__(self, user_settings: str = None) -> None:
+        """ The initialize function downloads the index files form GDAC and 
+            stores them in the proper directories defined in the 
+            DownloadSettings class. 
+
+            :param: user_settings : str - An optional parameter that will be used
+                to initialize the Settings classes if passed. Should be the 
+        """
+        self.download_settings = DownloadSettings(user_settings)
+        self.source_settings = SourceSettings(user_settings)
+        if self.download_settings.verbose: print(f'Starting initialize process...\n')
+        
+        if self.download_settings.verbose: print(f'Your current download settings are: {self.download_settings}')
+        if self.download_settings.verbose: print(f'Your current source settings are: {self.source_settings}')
+
+        # Check for and create subdirectories if needed
+        if self.download_settings.verbose: print(f'Checking for subdirectories...\n')
+        self.__initialize_subdirectories()
+
+        # Download files from GDAC to Index directory
+        if self.download_settings.verbose: print(f'\nDownloading index files...\n')
+        for file in self.download_settings.index_files:
+            self.__download_index_file(file)
+
+        # Load the argo_synthetic-profile_index.txt file into a data frame
+        if self.download_settings.verbose: print(f'\nTransfering index files into data frames...')
+        self.synthetic_index = self.__load_synthetic_dataframe()
```

Comment:
sprof_index (here and anywhere else)

Reply:
![image](https://github.com/lovelysandlonelys/OneArgo-Python-Dev/assets/91633941/64db8939-b1bd-4963-ae96-381d38b6006a)
Fixed here and in functions.